examples/counters/src/reducers/countersPairReducer.ts

```diff
@@ -0,0 +1,31 @@
+import { Action } from 'redux';
+import { buildReducer, buildUnwrapper } from 'prism';
+
+import counterReducer, { CounterState, initialState as counterInitialState } from './counterReducer';
+
+export interface CountersPairState {
+  top: CounterState,
+  bottom: CounterState
+};
+
+const initialState : CountersPairState = {
+  top: counterInitialState,
+  bottom: counterInitialState
+};
+
+export default buildReducer<CountersPairState>({
```

Comment:
Function currying prevents type inference in this case. Generic parameter `CountersPairState` could have been inferred from `initialState`.